PRCS/prc-1.md

```diff
@@ -0,0 +1,133 @@
+  id: string             // Game ID
+  name?: string          // Optional Game Name
+  version?: string       // Optional Game Version
+}
+```
+
+Data Validity
+For cross-game on-chain interoperability, this field allows the requester to confirm when the data is valid, and which is the reference blockchain.
+```
+interface Validity = {
+  block: number;         // Data Block height (0 always valid)
+  caip2: string;         // CAIP-2 blockchain identifier 
+  time: string;          // Optional Date ISO8601 YYYY-MM-DDTHH:mm:ss.sssZ
+}
+```
+
+Player Info
+``` 
+interface Player = {     
+  wallet:                // e.g., addr1234... or 0x1234..,
+  walletType?: 'cardano' | 'evm' | 'polkadot' | 'algorand' | string  // (Optional) Wallet-type
+  userId?: string;       // (Optional) User ID for a specific player account.
+                         // This value should be immutable and define a specific account,
+                         // as the wallet might be migrated or updated.
+  userName?: string;     // (Optional) Player Display Name
+}
+```
+
+## Specification
+
+### Get All Available Achievements
+`{BASE_URL}/achievements/public/list`
+
+Optional: Subset of achievements by category
+`{BASE_URL}/achievements/public/list?category=Silver`
+
+Optional: Subset of active achievements
+`{BASE_URL}/achievements/public/list?isActive=true`
+
+
+```
+interface AchievementPublicList extends Game, Validity {
+    achievements: {
+      name: string;                     // Unique Achievement String
+      score?: number;                   // Optional: Relative Value of the Achievement
+      category?: string;                // Optional: 'Gold' | 'Diamond' | 'Beginner' | 'Advanced' | 'Vendor'
+      percentCompleted?: number         // Percent of players that have unlocked the achievement 
+      isActive: boolean                 // If achievement can be unlocked at the time. 
+      displayName: string;              // Achievement Display Name
+      description: string               // Achievement Description
+      spoiler?: 'all' | 'description';  // Hide entire achievement or description if not completed
+      iconURI?: string;                 // Optional Icon for Achievement
+      iconGreyURI?: string;             // Optional Icon for locked Achievement
+      startDate?: string                // Optional Date ISO8601 YYYY-MM-DDTHH:mm:ss.sssZ
+      endDate?: string                  // Optional Date ISO8601 YYYY-MM-DDTHH:mm:ss.sssZ
+    }[];
+}
+```
+
+### Get completed Achievements for Wallet or Token
+`{BASE_URL}/achievements/wallet/:wallet`
+* wallet: Wallet address
+
+`{BASE_URL}/achievements/erc/:erc/:cde/:token_id`  
+`{BASE_URL}/achievements/erc/:erc/:cde/:token_id/:wallet_id`  
+
+e.g., /achievements/erc/erc1155/0/10/
+or /achievements/erc/erc721/1/20/0x1234
+* erc: Any supported ERC standard by the game, such as erc721, erc6551, or erc1155.
+* cde: Chain Data Extension identifier, might be used by the game to track identify the contract
```

Comment:
do we want to use a CDE id instead of name? IDs feel like stable because in theory you can add/remove primitives from your application (although it requires going in and manually changing stuff like we did for Tarochi)

Reply:
Updated to name 👍 it feels better